src/context/WalletProvider/WalletProvider.tsx

```diff
@@ -98,7 +98,8 @@ export interface InitialState {
   keyring: Keyring
   adapters: Adapters | null
   wallet: HDWallet | null
-  type: KeyManager | null
+  modalType: KeyManager | null
+  connectedType: KeyManager | null
```

Comment:
We now store what type of wallet is connected so we can show the appropriate "connected" menu, instead of assuming that wallet related to the last opened modal is what's connected.